recipes/mimalloc/all/test_package/conanfile.py

```diff
@@ -0,0 +1,91 @@
+from conans import ConanFile, CMake, RunEnvironment, tools
+import os
+
+
+class MimallocTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        # No override:
+        if not self.options["mimalloc"].override:
+            self._test_files = ["mi_api"]
+
+        # Visual Studio overriding:
+        elif self.settings.compiler == "Visual Studio" and self.options["mimalloc"].shared:
+            self._test_files = ["include_override", "mi_api"]
+        elif self.settings.compiler == "Visual Studio" and not self.options["mimalloc"].shared:
+            self._test_files = ["include_override", "mi_api"]
+
+        # Non Macos injected override:
+        elif self.settings.os != "Macos" and \
+             self.options["mimalloc"].override and \
+             self.options["mimalloc"].shared and \
+             self.options["mimalloc"].inject:
+            self._test_files = ["no_changes"]
+
+        # Could not simulate Macos preload, so just ignore it:
+        elif self.settings.os == "Macos" and \
+             self.options["mimalloc"].override and \
+             self.options["mimalloc"].shared and \
+             self.options["mimalloc"].inject:
+            self._test_files = []
+
+        # Unix-like non injected override:
+        else:
+            self._test_files = ["include_override", "mi_api"]
+
+        cmake = CMake(self)
+        cmake.definitions["BUILD_NO_CHANGES"] = "no_changes" in self._test_files
+        cmake.definitions["BUILD_INCLUDE_OVERRIDE"] = "include_override" in self._test_files
+        cmake.definitions["BUILD_MI_API"] = "mi_api" in self._test_files
+        cmake.configure()
+        cmake.build()
+
+    @property
+    def _lib_name(self):
```

Comment:
It would be nice to remove duplication, but I don't know if threr is a way to do it at the moment 😢

Reply:
Indeed. There's the [python requires](https://docs.conan.io/en/latest/extending/python_requires.html) thing, but I think it would be too much in this case...

Reply:
Agree